iree/compiler/Dialect/HAL/Transforms/test/pack_allocations.mlir

```diff
@@ -0,0 +1,100 @@
+// RUN: iree-opt -split-input-file -iree-hal-target-backends=dylib-llvm-aot -iree-hal-pack-allocations -cse -canonicalize %s | IreeFileCheck %s
+
+// CHECK-LABEL: @packStatic
+// CHECK-SAME: %[[ALLOCATOR:.+]]: !hal.allocator
+func @packStatic(%allocator: !hal.allocator) ->
+    (index, index, index, index, index, index, index) {
+  %c100 = constant 100 : index
+  %c200 = constant 200 : index
+  %t:7 = hal.allocator.pack<%allocator : !hal.allocator> slices({
+    [0, 1] = %c100,  // +0
+    [1, 2] = %c100,  // +112 (100 align 16)
+    [2, 3] = %c100,  // +0 (reuse [0, 1])
```

Comment:
more checking my understanding...

* the first slice lives from `[0, 1]` while this slice lives from `[2, 3]`
* since there is no overlap, this slice can reuse the 100 bytes of memory used for slice 1, which has no offset from the start of the slab

am I interpreting this correctly?

Reply:
yep!